Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiSwitcherManager.cpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ViewportUi/ViewportUiSwitcherManager.h>
+#include <AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.h>
```

Comment:
suggestion: We probably don't want this to have to include `EditorTransformComponentSelection` - it might be better to move the definition of what is currently `SwitcherCluster` outside of `EditorTransformComponentSelection` or just move all this logic inside `EditorTransformComponentSelection` for now and extract the parts we need later